source/postprocess/visualization/seismic_melt.cc

```diff
@@ -0,0 +1,372 @@
+            solution_names.push_back(property_names[i]);
+            std::replace(solution_names.back().begin(),solution_names.back().end(),' ', '_');
+          }
+        return solution_names;
+      }
+
+      template <int dim>
+      std::vector<DataComponentInterpretation::DataComponentInterpretation>
+      MeltSeismicProperties<dim>::
+      get_data_component_interpretation () const
+      {
+        std::vector<DataComponentInterpretation::DataComponentInterpretation> interpretation;
+        for (unsigned int i=0; i<property_names.size(); ++i)
+          {
+            interpretation.push_back (DataComponentInterpretation::component_is_scalar);
+          }
+        return interpretation;
+      }
+
+      template <int dim>
+      UpdateFlags
+      MeltSeismicProperties<dim>::
+      get_needed_update_flags () const
+      {
+        return  update_values  | update_q_points;
+      }
+
+      template <int dim>
+      void
+      MeltSeismicProperties<dim>::
+      evaluate_vector_field(const DataPostprocessorInputs::Vector<dim> &input_data,
+                            std::vector<Vector<double> > &computed_quantities) const
+      {
+        AssertThrow(this->include_melt_transport()==true,
+                    ExcMessage("'Include melt transport' has to be on when using melt transport postprocessors."));
+
+        const unsigned int n_quadrature_points = input_data.solution_values.size();
+        Assert (computed_quantities.size() == n_quadrature_points,    ExcInternalError());
+        Assert (input_data.solution_values[0].size() == this->introspection().n_components,   ExcInternalError());
+
+        // Set use_strain_rates to true since the compaction viscosity might also depend on the strain rate.
+        MaterialModel::MaterialModelInputs<dim> in(input_data,
+                                                   this->introspection(), true);
+        MaterialModel::MaterialModelOutputs<dim> out(n_quadrature_points, this->n_compositional_fields());
+        MeltHandler<dim>::create_material_model_outputs(out);
+
+        this->get_material_model().evaluate(in, out);
+        MaterialModel::MeltOutputs<dim> *melt_outputs = out.template get_additional_output<MaterialModel::MeltOutputs<dim> >();
+        AssertThrow(melt_outputs != NULL,
+                    ExcMessage("Need MeltOutputs from the material model for computing the melt properties."));
+
+        const unsigned int porosity_idx = this->introspection().compositional_index_for_name("porosity");
+
+        for (unsigned int q=0; q<n_quadrature_points; ++q)
+          {
+            unsigned output_index = 0;
+            for (unsigned int i=0; i<property_names.size(); ++i, ++output_index)
+              {
+                double porosity = std::max(in.composition[q][porosity_idx],0.0);
+
+                if (porosity < 0)
+                  porosity = 0.0;
+
+                double Vp = 0.0;
+                double Vs = 0.0;
+                double contiguity = 0.0;
```

Comment:
You can just initialize the contiguity as 1.0 here, then you don't need the statement in line 188.